File: Utility/src/main/java/com/projects/bigdata/utility/Utility.java
package com.projects.bigdata.utility;

import java.io.FileNotFoundException;
import java.io.IOException;
import java.io.InputStream;
import java.util.Arrays;
import java.util.List;
import java.util.Properties;
import java.util.concurrent.ExecutorService;
import java.util.concurrent.TimeUnit;

import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

public final class Utility {
	private static final Logger logger = LoggerFactory.getLogger(Utility.class);
	
	public static final List<String> Symbols = Arrays.asList("UBSN", "CSGN", "AAPL", "GOOG", "MSFT", "ACN", "AGN");
	
	public static Properties getApplicationProperties(final String propertiesFileName) throws FileNotFoundException, IOException {
		final Properties p = new Properties();
		try(final InputStream inputStream = ClassLoader.getSystemResourceAsStream(propertiesFileName)) {
			p.load(inputStream);
			return p;
		}
	}
	
	public static void shutdownExecutorService(final ExecutorService es, long timeout, TimeUnit timeUnit) throws InterruptedException {
		es.shutdown();
		if (!es.awaitTermination(timeout, timeUnit))
			es.shutdownNow();
		logger.info("ExecutorService shut down.");
	}
	
	public static void sleep(final TimeUnit timeUnit, long timeout) {
		try {
			timeUnit.sleep(timeout);
		} catch (final InterruptedException e) {
			logger.warn("Sleep interrupted.", e);
		}
	}
}

File: Utility/src/main/kotlin/com/projects/bigdata/utility/Utility.kt
@file:JvmName("Utility")

package com.projects.bigdata.utility

import org.slf4j.LoggerFactory
import java.util.*
import java.util.concurrent.ExecutorService
import java.util.concurrent.TimeUnit

val Symbols: List<String> = listOf("UBSN", "CSGN", "AAPL", "GOOG", "MSFT", "ACN", "AGN")

private val logger = LoggerFactory.getLogger("com.projects.bigdata.utility.Utility")

fun shutdownExecutorService(es: ExecutorService, timeout: Long, timeUnit: TimeUnit) {
    es.shutdown()
    if (!es.awaitTermination(timeout, timeUnit))
        es.shutdownNow()
    logger.info("ExecutorService shut down.")
}

fun getApplicationProperties(propertiesFileName: String): Properties =
        ClassLoader.getSystemResourceAsStream(propertiesFileName)!!.use {
            val p = Properties()
            p.load(it)
            return p
        }

fun sleep(timeUnit: TimeUnit, timeout: Long) = try {
    timeUnit.sleep(timeout)
} catch (e: InterruptedException) {
    logger.warn("Sleep interrupted.", e)
}